src/status_im/ui/market/view.cljs

```diff
@@ -0,0 +1,98 @@
+    [status-im.common.home.top-nav.view :as common.top-nav]
+    [status-im.common.refreshable-flat-list.view :as refreshable-flat-list]
+    [status-im.ui.market.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn swap-header
+  []
+  (let [theme             (quo.context/use-theme)
+        operable-accounts (rf/sub [:wallet/operable-accounts])]
+    [rn/view {:style (style/swap-header-container theme)}
+     [quo/text
+      {:weight :semi-bold
+       :size   :heading-1
+       :style  (style/market-header-text theme)}
+      (i18n/label :t/market)]
+     [quo/button
+      {:size      32
+       :icon-left :i/swap
+       :on-press  #(rf/dispatch [:app.wallet/start-swap operable-accounts])}
+      (i18n/label :t/swap)]]))
+
+(defn sort-header
+  []
+  (let [theme (quo.context/use-theme)]
+    [rn/view {:style (style/sort-header-container theme)}
+     [quo/text
+      {:size  :paragraph-2
+       :style (style/sort-text theme)}
+      (i18n/label :t/market-cap)]
+     [quo/icon :i/arrow-down
+      {:size            12
+       :color           colors/neutral-50
+       :container-style {:margin-left 4}}]]))
+
+(defn token
+  [token-id]
+  (let [{:keys [price percentage-change market-cap short-name token-rank token-name image]}
+        (rf/sub [:ui.market/token token-id])]
+    [quo/market-token
+     {:token-short-name  short-name
+      :token-rank        token-rank
+      :token-name        token-name
+      :market-cap        market-cap
+      :price             price
+      :token-image       image
+      :percentage-change percentage-change}]))
+
+(defn ensure-tokens-updating
+  [event]
+  (let
+    [visible-tokens (-> event
+                        .-viewableItems
+                        js->clj)
+     token-at-top   (-> visible-tokens
+                        first
+                        (get "item"))
+     tokens-count   (count visible-tokens)]
+    (rf/dispatch [:app.market/ensure-tokens-updating token-at-top tokens-count])))
+
+(defn view
+  []
+  (let [token-ids                      (rf/sub [:ui.market/leaderboard])
+        tokens-loading?                (rf/sub [:wallet/home-tokens-loading?])
+        [init-loaded? set-init-loaded] (rn/use-state false)]
+    (rn/use-mount
+     #(rf/dispatch [:app.market/load-leaderboard]))
+    (rn/use-unmount
+     #(rf/dispatch [:app.market/close-leaderboard]))
+    (rn/use-effect
+     #(when (and (boolean? tokens-loading?) (not tokens-loading?) (not init-loaded?))
+        (set-init-loaded true))
+     [tokens-loading?])
+    [rn/view {:style (style/home-container)}
+     [common.top-nav/view]
+     [refreshable-flat-list/view
+      {:refresh-control           [rn/refresh-control
+                                   {:refreshing (and tokens-loading? init-loaded?)
+                                    :colors     [colors/neutral-40]
+                                    :tint-color colors/neutral-40
+                                    :on-refresh #()}]
```

Comment:
This anonymous function `#()` will cause `refresh-control` to always be reprocessed by reagent.